### DISCUSSION TITLE
关于私密文件夹的哈希密码

### DISCUSSION THREAD

**Discussion by opoet** [spencerwooo/onedrive-vercel-index#87] [Ideas]
我注意到 #82 这个 pr 中提交的哈希密码功能和最后合并的 #83 中的哈希密码功能其实是不同的

-  #82 中实现的功能是将实际密码的 SHA256 哈希值存入`.password`文件中，然后将输入的密码转化为 SHA256 哈希来对比验证。
-  #83 中实现的功能是将`.password`文件中的值和输入的密码都转换为 SHA256 哈希来对比验证。

我的想法是将密码的哈希值存入`.password`文件中，并且在输入密码后将其转换为哈希值并存入 localStorage 中，再对比验证两个哈希值是否相同。请问这样是否有利于提高隐私文件目录的安全性，或者作者另有打算？

**Comment by spencerwooo** [comment]
噢，所以 #82 的本意是设置 `.password` 文件时需要手动 hash 密码并存进去，然后输入 token 输入的依旧是 hash 之前的明文？个人觉得这有点多此一举。

我原本是没有看到 #82 是如何校验 hash 之后的 token 的，以为原 author 忘记实现这部分了，并且原 PR 直接用了 Node.js 的 crypto module，这个不能直接在 browser 里面用，所以我直接重新写了，没让原作者进一步更新。

**Reply by spencerwooo** [reply]
总之 token 需要手动 hash 之后存入 `.password` 的这个操作非常 counterintuitive，所以我不准备这样实现。

**Reply by opoet** [reply]
确实不够《优雅》😕，期待更棒的方案
